### ISSUE TITLE
Gather feedback / investigate documentation required for `OrderBook` class

### ISSUE THREAD

**Issue by dennisocana** [CoinAlpha/hummingbot#4122]
## Why

We want to add to our developer docs information how `OrderBook` works. For example, we can document all the functionality how the strategies access the information stored in the orderbook that could be important to make decisions inside a strategy. Since this functionality is implemented in C, it's not easy to see using a Python IDE.

## What

List down common questions from the dev community and gather feedback to help us understand how we want to format the `OrderBook` documentation and what information will be useful.

**Comment by dennisocana** [comment]
Further research is required and this story needs to be revised. We should identify what common questions we are getting from the dev community that this task intends to address. We think the documentation will be easier to understand by putting only the information relevant to readers.

**Comment by dennisocana** [comment]
Question from @CrimsonJacket:

Documenting the functions within the `OrderBook` to update it is pretty straightforward but how in-depth do we want it to be documented? For example:

`c_apply_diff(...)`
- Applies the list of bids and asks to the orderbook
- params
-- `asks: List[OrderBookRow]`: List of asks
-- `bids: List[OrderBookRow]`: List of bids
-- `update_id: int`: New Update ID to be assigned to the order book

However, it would be important to data is encapsulated within the `OrderBookRow` and how we transform the orderbook diff messages(from exchange's WS API) to `OrderBookRow`.

So in short, are we going to have to document the entire pipeline, from raw messages -> `OrderBookRow` -> apply to `OrderBook`?

**Comment by dennisocana** [comment]
We're going to do a How-to style guide instead. Refer to the main documentation updates epic here: https://app.zenhub.com/workspaces/hummingbot-5f340755c5e17300167a8f30/issues/coinalpha/hummingbot/4162
